File: baekjoon/수학/소수상근수.py
def sosu(num):
    t = 2
    if num < 2:
        return False

    while t * t <= num:
        if num % t == 0:
            return False
        t += 1
    return True

def saangn(num, visit):
  result = 0
  for i in list(str(num)):
    result += int(i)**2
  if result == 1:
    return True
  elif result in visit:
    return False
  else:
    visit.append(result)
    return saangn(result, visit)



number = int(input())
sosuArr,visit = [],[]
for i in range(2,number+1):
  if sosu(i): sosuArr.append(i)

for i in sosuArr:
  if saangn(i,visit):
    print(i)
  visit = []

File: By-Swift/By-Swift/BaekJoon/소수상근수.swift
func sosu(_ num: Int) -> Bool {
  var t = 2
  while t * t <= num {
    if num % t == 0 {
      return false
    }
    t += 1
  }
  return true
}

func saangn(_ num: Int,_ visited: [Int]) -> Bool {
  var result = 0
  var visited = visited
  let splitNum = Array(String(num))
  
  for i in splitNum {
    result += Int(String(i))! * Int(String(i))!
  }
  if result == 1 {
    return true
  } else if visited.contains(result) {
    return false
  }
  visited.append(result)
  return saangn(result, visited)
}

let n = Int(readLine()!)!

for i in 2...n {
  let visited: [Int] = []
  if sosu(i) && saangn(i,visited){
    print(i)
  }
}